components/net/tests/http_loader.rs

```
@@ -68,15 +68,18 @@ fn assert_cookie_for_domain(
 }
 
 fn recv_http_request(devtools_port: &Receiver<DevtoolsControlMsg>) -> DevtoolsHttpRequest {
-    match devtools_port.recv().unwrap() {
-        DevtoolsControlMsg::FromChrome(ChromeToDevtoolsControlMsg::NetworkEvent(_, net_event)) => {
-            match net_event {
-                NetworkEvent::HttpRequest(req) => req,
-                NetworkEvent::HttpRequestUpdate(req) => req,
-                other => panic!("Expected HttpRequest but got: {:?}", other),
-            }
-        },
-        other => panic!("Expected NetworkEvent but got: {:?}", other),
+    loop {
+        match devtools_port.recv().unwrap() {
+            DevtoolsControlMsg::FromChrome(ChromeToDevtoolsControlMsg::NetworkEvent(
+                _,
+                net_event,
+            )) => match net_event {
+                NetworkEvent::HttpRequest(req) => return req,
+                NetworkEvent::HttpRequestUpdate(req) => return req,
+                _other => continue,
```

I'm not sure the changes to this function are the best way to.solve the test issue. Silently ignoring responses seems like it can hide problems.

---

I have tried different ways to get around this and can't seem to come up with a better solution for the test issue, what do you suggest I do instead sir?

---

The following change makes the test pass for me:
```rust
diff --git a/components/net/http_loader.rs b/components/net/http_loader.rs
index f03df194c34..3b08288f62f 100644
--- a/components/net/http_loader.rs
+++ b/components/net/http_loader.rs
@@ -990,8 +990,6 @@ pub async fn http_fetch(
         .try_code()
         .is_some_and(is_redirect_status)
     {
-        // Notify devtools before handling redirect
-        send_response_to_devtools(request, context, &response, None);
         // Substep 1.
         if response.actual_response().status != StatusCode::SEE_OTHER {
             // TODO: send RST_STREAM frame
diff --git a/components/net/tests/http_loader.rs b/components/net/tests/http_loader.rs
index 3a152130fa8..4474b80d42b 100644
--- a/components/net/tests/http_loader.rs
+++ b/components/net/tests/http_loader.rs
@@ -68,18 +68,16 @@ fn assert_cookie_for_domain(
 }

 fn recv_http_request(devtools_port: &Receiver<DevtoolsControlMsg>) -> DevtoolsHttpRequest {
-    loop {
-        match devtools_port.recv().unwrap() {
-            DevtoolsControlMsg::FromChrome(ChromeToDevtoolsControlMsg::NetworkEvent(
-                _,
-                net_event,
-            )) => match net_event {
-                NetworkEvent::HttpRequest(req) => return req,
-                NetworkEvent::HttpRequestUpdate(req) => return req,
-                _other => continue,
-            },
-            other => panic!("Expected NetworkEvent but got: {:?}", other),
-        }
+    match devtools_port.recv().unwrap() {
+        DevtoolsControlMsg::FromChrome(ChromeToDevtoolsControlMsg::NetworkEvent(
+            _,
+            net_event,
+        )) => match net_event {
+            NetworkEvent::HttpRequest(req) => return req,
+            NetworkEvent::HttpRequestUpdate(req) => return req,
+            other => panic!("Expected HttpRequest but got: {other:?}"),
+        },
+        other => panic!("Expected NetworkEvent but got: {:?}", other),
     }
 }

```

This PR makes us always send a response to the devtools, even when there's a redirect, so we no longer need to special case the redirect code path.

---

this still fails for me unfortunately , 
`---- http_loader::test_redirected_request_to_devtools stdout ----

thread 'http_loader::test_redirected_request_to_devtools' panicked at components/net/tests/http_loader.rs:101:22:
Expected HttpResponse but got: HttpRequest(HttpRequest { url: "http://localhost:60624/", method: GET, headers: {"accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "accept-language": "en-US,en;q=0.5"}, body: None, pipeline_id: (1234,5678), started_date_time: SystemTime { tv_sec: 1753951446, tv_nsec: 402759000 }, time_stamp: 0, connect_time: 0ns, send_time: 0ns, destination: Document, is_xhr: false, browsing_context_id: (1234,8765) })

`

---

Huh; I don't see this failure in debug unit tests, only when using the --release flag. That's fun! I'll keep poking at this.